src/main/kotlin/org/jetbrains/research/testspark/tools/kex/generation/KexProcessManager.kt

```diff
@@ -0,0 +1,350 @@
+                        KexMessagesBundle.get("unsupportedCodeTypeLine"),
+                        project,
+                        errorMonitor,
+                    )
+                    return null
+                }
+            }
+
+            // Disallow old java versions
+            val version = ToolUtils.getJavaVersion(javaExecPath)
+            if (version == null || version < 8) {
+                kexErrorManager.errorProcess(KexMessagesBundle.get("incorrectJavaVersion"), project, errorMonitor)
+                return null
+            }
+            Path(generatedTestsData.resultPath).createDirectories()
+
+            // Download kex executable if not present
+            if (!ensureKexExists()) {
+                kexErrorManager.errorProcess(KexMessagesBundle.get("invalidUrl"), project, errorMonitor)
+                return null
+            }
+
+            val cmd = KexSettingsArguments(
+                javaExecPath,
+                projectContext.cutModule!!,
+                target,
+                resultName,
+                kexSettingsState,
+                kexExecPath,
+                kexHome,
+            ).buildCommand()
+
+            val cmdString = cmd.fold(String()) { acc, e -> acc.plus(e).plus(" ") }
+            log.info("Starting Kex with arguments: $cmdString")
+
+            // run kex as subprocess
+            runKex(cmd, errorMonitor, indicator)
+
+            log.info("Save generated test suite and test cases into the project workspace")
+
+            preprocessGeneratedTestFiles(resultName, classFQN, generatedTestsData, projectContext, errorMonitor)
+        } catch (e: Exception) {
+            kexErrorManager.errorProcess(
+                KexMessagesBundle.get("KexErrorCommon").format(e.message),
+                project,
+                errorMonitor,
+            )
+            e.printStackTrace()
+        }
+
+        return UIContext(
+            projectContext,
+            generatedTestsData,
+            StandardRequestManagerFactory(project).getRequestManager(project),
+            errorMonitor,
+        )
+    }
+
+    private fun preprocessGeneratedTestFiles(
+        resultName: String,
+        classFQN: String,
+        generatedTestsData: TestGenerationData,
+        projectContext: ProjectContext,
+        errorMonitor: ErrorMonitor,
+    ) {
+        val report = IJReport()
+        val imports = mutableSetOf<String>()
+        val packageStr = classFQN.substringBeforeLast('.')
+
+        val generatedTestsDir = File(
+            "$resultName/tests/${classFQN.substringBeforeLast('.').replace('.', '/')}")
+        if (generatedTestsDir.exists() && generatedTestsDir.isDirectory) { // collect all generated tests into a report
+            for ((index, file) in generatedTestsDir.listFiles()!!.withIndex()) {
+                val testCode = file.readText()
+
+                if (file.name.contains("Equality") || file.name.contains("Reflection")) {
```

Comment:
These constants should be declared somewhere as statics. Also, it is better to specify their full name to not cause any unexpected behaviour